### DISCUSSION TITLE
really need help

### DISCUSSION THREAD

**Discussion by jt4242** [OpenSageTV/Community#138] [General]
I've used Sage for years but took it all apart the other day because I bought a new TV - Samsung SAMSUNG 50-Inch Class QLED 4K Q60D.  But I hate it - the UI is not intuitive.  Unfortunately I hadn't taken photos of my setup before the guy came to install so I can't remember what goes where.  My cable box is a comcast xi-6 and I have the Sage HD300.  But where does the hdmi from my cable box go?  and what goes into the HD300 Hdmi port and also what are the 2 USB ports for.  Sorry to be so blank but it's  been so many years since I set Sage up and my memory is shot!!!

I did have hdprs and splitters attached to the cable box but they started failing.

I don't know if anyone can help but it would be greatly appreciated - I've hooked it for the moment to the TV with the HDMI so I can play all my recordings - I loved the Sage interface.  But I'm unable to play live TV.  I also have the cable box attached to the TV with an HDMI cable so I can switch to that HDMI and watch live but it's just easier doing it all from Sage.

Also the cable box doesn't show up in my ADD NEW SOURCE in the source setup menu.  I sound so pathetic but it's hard when one's memory fails!!!!!!!!!!!

Thanks in advance for any help anyone can give me!!
julie

**Comment by LoisGNS** [comment]
The HD300 outputs to your tv through the HDMI port. The input to the HD300 is an ethernet cable to connect to your home network so it can get data from your Sage Server. 

I'm not familiar with your particular cable box, but if you want to use it as a tuner for Sage, you'll need at least two things: (1) a connection to whatever tuner attached to the computer that you use for Sage, and what you'll need - and whether it's possible - depends on the specifics of your cable box & tuner. (2) a way to control your cable box from Sage, such as a USBUIRT or other type of controller. Connecting the cable box directly to the HD300 will do nothing. If you have more HDMI outputs that you want to use than you have TV HDMI inputs, you can get a switchbox to expand your connections. 

My best recollection is that the USB ports on the HD300 are for processing system updates, and one might be for a wifi adapter to use instead of ethernet. 

If your cable box is connected through Sage, you should be able to use it to watch live TV, but if you want to watch one thing live and record something else at the same time you'll need a signal sources & tuner as you need simultaneous watching/recording sessions. Some boxes or tuners support multiple inputs, so count those as multiple tuners for this purpose if you have them. 

My own setup, as an example, involves a few USB tuners attached to my computer and to OTA antenna output, a couple of cards that each supports OTA input plus inputs to accept channel 3/4 cable box output. For those, I use the Comcast DCA boxes. All of those feed into my computer for recording, but can also be used live if I choose. I even have an ancient HDhomerun on the system, but it gets used only rarely as the tuners are pretty weak. The main TV has another Comcast box on one HDMI input and a Sage HD200 on another. I have some other Sage boxes attached to secondary TVs in the house. But for all of this to work, you need to have usable tuners & signal sources attached to your server directly or via LAN, as with the HDhomerun.

**Comment by jt4242** [comment]
I'll write lots more tomorrow but I'm fighting with it again - now none of the sage is working - if you go into the hdmi icons for each thing you have hooked up and, on my tv, hit the down arrow under each, there are editing choices - first on mine is edit device type - but I don't know what type of device the HD300 is!  choices are pc, cable box, satellite box, iptv, pvr stb, game, blue-ray, hd dvd, dvd, vcr, av receiver, camcorder, dvi, pc, tv, dma.

The hdmi output from the hd300 is going directly into an hdmi port on my TV and it was working fine for me to watch all my recordings then it started going on and off for short periods so I started trying to fix it and now I can't even get Sage to open on my screen - I'm a mess and so is my system.  I appreciate you writing and will get to the rest once I get sage up and moving again.  Going to do a reboot now to see if that will help.

**Comment by emveepee** [comment]
Not understanding if you mean you want to capture from the Comcast box you might want to read this thread https://github.com/OpenSageTV/Community/discussions/114

**Comment by jt4242** [comment]
That will be the next step - what I'm trying to do now is access all the
Sage recordings I have on my PC.  I start Sage and select the server.
Screen shows "connecting to server" and then nothing else happens - screen
stays black..  I have my HD300 connected to one of the HDMI ports on the TV
and it used to work that way.  I  am wondering if it's something to do with
my network connection?  But I've been trying everything - I really messed
it all up this time.

**Comment by wayner9** [comment]
Does your HD300 see your network and have a valid IP address on your network?  The boxes can sometimes be finicky starting up if they don't have a valid HDMI connection to a TV/monitor.  Try plugging the HDMI from the HD300 into the TV, turning to the correct TV input and then power cycling the HD300.  You should at least see a SageTV screen when the system is doing its inital bootup and then it should go to the SageTV standalone page if it doesn't automatically connect to your server.

You should be able to get things working and play back previously recorded content. That shouldn't be too hard.  But to Watch live TV and record from that cable box may take a bit more work.

But the good thing is that the Comcast Xfinity system is actually very good.  You can record on their cloud system and you should also be able to access these recordings on mobile devices and via a web portal.  And any TV on your house with a Comcast box can see the TV shows.  And the voice control on the remote is excellent.  You might want to consider getting the SageTV system and running to watch your older content, and then use the Comcast box for live TV and new recordings.  You also may be able to use an app on devices like Roku or FireTV to access the Comcast content.

**Reply by jt4242** [reply]
OMG!!!  After days of doing the same thing - rebooting restarting modem etc, today it worked and I can see sage on my TV!!  Only problem now is that I'd afraid to turn it off and lost it maybe!!!!!!  I'm making an Acronis backup of my C drive and hope that if I do lose it that a restoration of the drive will give me what I need!!!

**Reply by jt4242** [reply]
Now I'm on to the next step - attaching the HDPVR and splitter so I can record the output of my cable box and I cannot remember where the splitter goes - is the in HDMI from the cable box?  And where do the 2 HDMI outputs go?  I do have a splitter and USB-UIRT so I've got the equipment but not the memory!!!  thanks again for any help you can give!!

**Reply by wayner9** [reply]
Which HD-PVR do you have?  Does it have HDMI inputs or Component inputs?  If it has Component then you need an HDMI to Component adapter.  You don't really need a splitter - the signal should go from your cable box to your HD-PVR.  Then there is an output from the HD-PVR that you can hook up to a TV or monitor if you want but that isn't necessary if you don't want it.  Then you need to control the channel changing on the cable box using IR from your HD-PVR or whatever.  The remotes that come with these boxes use Bluetooth but the boxes also accept IR commands.

**Reply by jt4242** [reply]
Hi again and thanks for writing!!  I have a few of them - 1 x HD-PVR 2 GE PLUS AND the others HDPVR2 - I also have an Hd-PVR60 but I've never been able to get that one to work!  I have always used the HDMI inputs.  I thought we needed the splitter for the HDCP when we wanted to record from sources like cable boxes.  So my signal goes from the Comcast (Helix here in Canada) to the HDPVRs that I use for recording I think..  But I can't remember if the splitter goes between them or on the output from the HD-PVR. It's just that there are 2 outputs and I can't remember where they went!  One must go into my TVs HDMI I think.  But the other I have no idea.

Another one of those instances when I so miss the old forums - I've been searching but have not found postings I know I used to use.  As for changing the channel on the cable box, I've used USB-UIRTs for years and still have a few.

Thanks for your help!  Maybe I'll have a dream some night and see my old hookups!!
julie

**Reply by wayner9** [reply]
You may be right about needing the splitter to overcome the HDCP on the boxes.  I forgot about that as I use a HDMI-component adapter that doesn't pay attention to HDCP.

The HD-PVR2 takes HDMI inputs so that should be good.  I have the same cable box as you - I am with Rogers Cable in Toronto and have an Xi-6 as well.  You should be able to just plug an HDMI cable from the Xi6 to the splitter to the HD-PVR2.  The USB-UIRT should work but you might have to teach the USB-UIRT the number codes and the enter code.  To do that you might have to put the remote into IR mode.  See this thread on how to do that: https://communityforums.rogers.com/t5/Ignite-TV/Ignite-Tv-Remote-in-IR-mode-only/td-p/451237

**Reply by jt4242** [reply]
I was using 2 Xi6 boxes with 2 HDPVRs before but haven't for a few months - it was great being able to record 2 shows at once as they seem to have all the shows I like on the same night!!  And I hate commercials so really need to run them through comskip!!!

So HDMI from the Xi6 to the splitter and 1 output to the HDPVR but what about that second output - my memory is blank.  And the spinning circle on my screen is tortuous now and I can't remember what I used to do to get rid of it and, sadly, can't search the forum!!! 

I used USB-UIRTs before (in fact they are still taped to the front of my box) - they always worked fine.  so I'm not sure what you mean about the number codes and enter code!!  I thought I had just used them to change channels but what you say makes sense?!  You're great and thank you for responding - I liked the HDMI better because it was easier to connect - I'm not sure what the component adapter is

**Reply by wayner9** [reply]
I don't think you don't need to attach the second output of the splitter to anything although you could attach it to a TV or monitor if you wanted to.

The USB-UIRT sends infrared codes to change channels on the cable boxes.  But each cable box has different codes.  You probably currently are setup for boxes like the SA4250HD or other Cisco/SA boxes.  You now need to use codes for the Comcast Xfinity boxes.  You might be able to download a file from somewhere, but I believe you can also teach the codes by pointing the remote at the USB-UIRT and pressing buttons.  But the Xfinity remote does not use IR by default.  So you have to turn that on using the link I provided.  I believe that there is a folder off of  the SageTV folder that has files that could be called something like sa.ir.  These are the ir codes.

This thread in the old forums may have the file that you need so that you don't have to teach the codes:  https://forums.sagetv.com/forums/showthread_t_63276.html?t=63276

**Reply by jt4242** [reply]
Maybe that's why I can't remember what the 2nd output was for - it was for nothing!

quickie question - on the main menu page there are icons on the top right just to the left of the date - one of them - the red and green verticle one and when it's too red the java heap is too big.  I used to press some key to reduce it but can't remember which one - do you know?

Also since getting things working, every time I do anything on the menu (start/stop a recording, go from topic to topic, the stupid spinning circle spins for ages and it gets so tedious having to wait all the time - any ideas?

thanks and hope Toronto weather is as gorgeous as ours in Montreal is!!

**Comment by wayner9** [comment]
What you are talking about with the green/red bar is the Java heap size.  This can be an issue if you have multiple clients running.  I have about 9 TVs with SageTV HD300 or HD200 extenders.  With the 32 bit version of SageTV If you had more than two or three extenders then Sage could crash.  But that problem went away with the 64 bit version.  You can see what version you are running by going to Setup and then System Information.  My system says SagetV V9.2.8.1 (64-bit).  If you scroll down it should say JVM Heap Size.  Mine shows "JVM Heap Size (Used/Total/Max): 1235MB/1568MB/2147MB" .  If I remember correctly there was a limit of 1024MB with the 16 bit version, but with the 32 bit version you can go much larger.  2048 or 3072 should be plenty - for fun I once connected about 15 SageTV miniclients and didn't have problems.

To change Java Heap size see this thread from the old forums, https://forums.sagetv.com/forums/showthread_t_50327.html?t=50327
Or other threads.

Not sure why you get the spinning circle, but more Java Heap size might help.  Having your system files on a fast SSD rather than a spinning hard drive also helps.

I am assuming that you are running SageTV on Windows.  My main SageTV systems run on unRAID Linux, but I do have SageTV  on one Windows PC, but I don't really use it much.

**Reply by jt4242** [reply]
Yes, I knew it was the java heap size but there was some key I pressed on my remote that brought it right down and I can't remember what key (or combo?).  It was so easy!  and I do need it as my heap gets pretty big. I did increase the java heap size a couple of days ago (not a lot) but it made no difference.

My devices are minimal now - 1 hd300 and 1 helix box. 
I have 32GB RAM

My Sage is SagetV  V9.2.4.920 (64-bit)
JVM heap size 712MB/2580MB/7636MB

I used to play a lot with the size - it was at 2000 but I changed it to your recommendation 3072 (WAS THAT HEXADECIMAL - THAT'S WHAT i DID)
(in case the number had to be a multiple of some number????

Windows 10

We'll see how that goes - I can't keep interrupting my husband's TV watching this way!!!!!!!!!!!  haha

not sure how to reply to you - I did it by email a couple of times but don't see the content on the forum.

Strange other problems - the menus for example on placefhifter client on my PC there is the bar under each recording that is green or green and red or shows the comskip markings.  I dont' have that anymore on my TV set and every category opens and I have to reset the sorting of each genre.  Plus my recordings, even though the genres I gave them are fine the are a jungle on the tv interface mixing up all the genres.  For example one of the menu categories I add years ago is Home where I put home improvement type shows - now it's full of Drama, home shows, movies and many others - a big mess.

**Reply by jt4242** [reply]
I did find: I believe if you go to System Info in Sage and hit the Delete key it cleans up the Java memory.  -This was in a posting from NYPlayer in 2012 (link:  https://forums.sagetv.com/forums/showthread_t_58798_page_2.html?t=58798&page=2).

Now the only problem is where is the delete key!!!?????

**Reply by wayner9** [reply]
There is a delete key on the silver SageTV extender remotes - it is near the bottom right corner of the remote.  The delete key on a keyboard will probably work in Placeshifter.
Those green bars are set in UI settings that are different for each client.  Somewhere in the setup menu, or in the menu options you can set those.  I am using the Gemstone UI - not sure if you are using the default UI or not.

I know what you mean about interrupting others - I have a 22 year old autistic daughter.  She watches SageTV at times, so I have to play around when she is not watching.  I used to do major work after she and the rest of the family went to bed and at times when I would screw things up I would be up until 3-4 in the morning to fix it.  Tabernac! As you would say in Quebec.

**Reply by jt4242** [reply]
OMG -   I searched my silver remote and didn't see it - thank you so much!!  Now to just figure out what I mapped it to on my universal remote! I'm using the standard interface - in this new incarnation it's my only choice!  though I did amend the menus a lot with ADM (another dynamic menu) when I built the system 7 years ago but can't get everything back to normal.  You're talking about me this past week - up till 3:30 every night!  It's been a long time since I've had something so challenging to get my teeth into!  The nice part is that he starts my day by bringing me an espresso - very welcome after the lost night!

**Reply by jt4242** [reply]
System much speedier now - thanks for the heap tip - I guess the memory of that 1024 maximum was buried deep in my brain!!

**Reply by jt4242** [reply]
I still have a problem - I wanted to upload a photo of my screen but can't so will try to explain.  My main menu has 10 categories down the left side - some added by me but I can't remember how.  For example, I have one called Old Shows, One called Cinema, one names John etc.  I created genres with the same names so they'd go the right place but I don't remember how I connect the genre to the menu item so the directories for each of these is a mess of different genres since my disaster.  If you had an email I could sent it to you - my name is jt4242 and I use that common email service that begins with a G so maybe you could write me?

**Reply by LoisGNS** [reply]
Thanks for the tip about using the Delete key to reduce the heap. For ages, I've been watching it get mostly red, and the system getting very laggy, and all I could do to clear things out & defeat the lags was go to my server and either stop/start the Sage server, or reboot the system. Next time I get lags, I'll try the system setting --> delete and see if it helps. It does push the heap down when I use it, but it comes back up - at least part of the way - almost immediately.

**Reply by wayner9** [reply]
@LoisGNS - Can you increase your heap size?  That seems to help keep the system running well. I typically have 4-5 extenders connected and don't run into problems.

**Reply by jt4242** [reply]
to LoisGNS - glad it helped - now I wish you could help me - since my computer disaster I've forgotten how to hook up my OTA antenna output using HDHomrun boxes.  My memory is so blanked out that I can't even remember how I used to test the antenna strength and I've been using them for years and years.  This is one of those discussions where I wish we could email each other - if you're open to that you can email me at jt4242 at the universally known email - starts with a g!
julie

**Reply by LoisGNS** [reply]
@wayner9, I have fiddled with my heap size, but it didn't seem to make much difference, so I stopped messing with it.
@jt4242, I only have minimal experience with HDHR boxes. I have one, but it's so weak as a tuner, that it's last on the "merit" list. Most of the work was getting drivers, etc., installed, and it's been a while. I think the HDHR software has some sort of a way to check antenna strength, but what I usually do for that is to use Sage: go to video sources area, go to the tuner to be checked, e.g. HDHR, and try to tune a channel that you should be able to receive. Watch the tuning-strength bar to assess. 

Also probably best to discuss here, so others can benefit from whatever we figure out.

**Reply by wayner9** [reply]
You can see the signal strength when you are in the channel setup screen.  Select the channel, click preview this channel and the audio and video should start in a second or two and it will show signal strength.

**Comment by jt4242** [comment]
Months later!!  I have Sage working beautifully now thanks to the help from you people here and also to finally remembering that I had to connect the HD300 to the modem with an internet cable!

But the one thing I still cannot do is get the USB UIRT connected so I wondered if, by chance, anyone reading this could give me some tips - It would be great if someone who lived in the Montreal area was on this forum!!!  I have Videotron and the Helix box is most annoying - it goes into a screensaver and then nothing records except the screensaver!!  I really have to get the USB UIRT working so I can send the box a signal every hour and keep it alive!

Hope someone is a miracle worker!!!
julie

**Reply by wayner9** [reply]
Have you been able to use the USB-UIRT to learn commands from the remote control for your TV box?  You will have to switch the remote into IR mode,.  Once you do that then I think SageTV might allow you to learn the IR codes from the remote control.  Here is how to put the remote in IR mode:
"The XR15 works a bit differently (it doesn't have a Setup button) but you should still be able to put that remote into IR mode by pressing and holding the A (triangle) and D (diamond) buttons simultaneously until the status light turns green, then punch in code "9 8 7"".  

With the USB UIRT there are a couple of companion windows programs.  One is called uutx.exe which is for transmitting codes and there is another one that is for learning codes.  I don't remember the name.  Contact them at support@usbuirt.com.  Jon Rhees is the guy that makes these and he used to be active on the old SageTV forums to help out.

Assuming that you can either find the IR codes on the internet or learn them then the way to keep the boxes alive is to send the back arrow command to the box every four hours.  This should keep the box alive.  You would do this by setting up a Scheduled Task in Windows.  You would then run a task that sends a command like :"c:\Program FIles\USBUIRT\uutx.exe 00ffaacc" 

Then you probably want to change your remote back to Bluetooth mode.

Maybe somebody else remembers the SageTV IR learning process with SageTV as I haven't used IR in a long time, and I use Unraid Linux.

One other way of learning the remote codes where you don't have to change your Videotron remote is to get a Harmony remote and set it up for the Comcast Xfinity box.  This is what I did as I have some old Harmony IR remotes lying around as I used to use them.

**Reply by wayner9** [reply]
And you can find the uutx.exe program on this page https://www.etcwiki.org/wiki/Uutx.exe_command_line_for_usb-uirt

**Reply by wayner9** [reply]
And the lrnhelper.exe file that you can use to learn files is linked to on this page.
http://usbuirt.com/phpBB3/viewtopic.php?f=2&t=5943

**Reply by emveepee** [reply]
It looks like Videotron is also licensing the Comcast hardwareand I posted the key codes here https://github.com/OpenSageTV/Community/discussions/114#discussioncomment-8959807 and you would need to send a KEY_BACK to keep the box awake. I don't bother, unless you record shows longer the the screen timeout and you can send the KEY_BACK to start your tuning string.   For me with Rogers I was able to set the screensaver to 4 hours not 1 hour so  maybe the software is different.

Also assuming  it is like the Rogers and Comcast boxes you don't need to turn BT mode off the box always understands BT and IR.

**Reply by jt4242** [reply]
Sorry but what is a Key_Back?  I do have my earlier .ir files but what I cannot remember is where I attached them to the box - and yes it is the Comcast xi6.  I had it taped to the front but now can't see where the receptor of the signal is.  My screensaver delay is set to 5 hours but there are a couple of other messages that pop - up sorry I can't remember but there is the reboot one every night and I'm sure one other.

**Reply by emveepee** [reply]
KEY_BACK would be the IR code to the left arrow on your remote.  The issue would be figure out how to sending it with USB UIRT , I use the LIRC codes.   KEY_BACK cancels a few message including not subscribed but I don't get a nightly reboot message, my uptime for the box is months now.

**Reply by JustFred51** [reply]
For USB-UIRT, learning the IR codes is pretty easy, since the capability is built-in to Sage. No need to use any external programs.  See pg. 166 of the SageTv Manual, which can be downloaded from http://download.sage.tv/SageTV_Manual.pdf.

The biggest issue with setting up USB-UIRT is that it's installer incorrectly installs the 32-bit version of the driver for 64-bit Windows.  The correct 64-bit version of C:\Windows\System32\uuirtdrv.dll is 278 KB.  The 32-bit version is 236 KB.  See: https://forums.sagetv.com/forums/showthread_t_65754_page_10.html?t=65754&page=10, post #191, although it shouldn't be necessary to manually copy the uu_irsage.dll file.

**Comment by jt4242** [comment]
I had meant to also thank Wayner9 especially for all the help he gave me!!
